scripts/g.extension/g.extension.py

```diff
@@ -1555,7 +1569,7 @@ def download_source_code(
             response = urlopen(url)
         except URLError:
             # Try download add-on from 'master' branch if default "main" fails
-            if branch == "main":
+            if branch is None or branch == "":
```

Comment:
Can be written as `if branch:` but I think the logic does not make sense. See my comments about trunk. Here perhaps result of having main as the default for the option.

Reply:
See: ca14583

Reply:
Can you be more specific on how you addressed this? The condition is still unnecessary complicated and I don't understand how it fits to the code which follows. Perhaps update the comment.